### PR TITLE
Include libspeechd for SpeechSynthesis support

### PR DESCRIPTION
**Pull request by hypevhs**
Electron apps may want to use the SpeechSynthesis part of the Web Speech API.

On Linux, this is gated behind the `--enable-speech-dispatcher` flag. If enabled, Chromium will attempt to dynamically load the Speech Dispatcher library `libspeechd.so.2`. If not present, speech is disabled. [Chromium source code for proof](https://source.chromium.org/chromium/chromium/src/+/refs/tags/122.0.6261.111:content/browser/speech/tts_linux.cc;l=251). Right now this library isn't included inside the Flatpak, even though the host may be fully configured with Speech Dispatcher.

This PR builds and includes the library (but not the daemon) so that TTS stuff can easily work as long as the consuming app passes the `--enable-speech-dispatcher` flag and has sandbox perms for `--filesystem=xdg-run/speech-dispatcher`.

This is useful for flathub/dev.vencord.Vesktop#20, though we thought it'd make sense to upstream it for other apps' benefit.

### CODE REVIEW

File: org.electronjs.Electron2.BaseApp.yml [hunk 1]
```diff
@@ -94,3 +94,46 @@ modules:
...
+  - name: libspeechd
+    config-opts:
+      - --disable-static
+      - --with-ibmtts=no
+      - --with-kali=no
+      - --with-baratinoo=no
+      - --with-voxin=no
+      - --without-flite
+      - --disable-python
+      - --with-systemdsystemunitdir=/app/lib/systemd/system/speech-dispatcherd.service
```
Comment (bbhtt) [inline org.electronjs.Electron2.BaseApp.yml]:
```suggestion
      - --with-systemdsystemunitdir=''
```

This is not really useful in the sandbox. You have to start it from your host.

File: org.electronjs.Electron2.BaseApp.yml [hunk 2]
```diff
@@ -94,3 +94,46 @@ modules:
...
+      - /include
+    sources:
+      - type: archive
+        url: https://github.com/brailcom/speechd/releases/download/0.11.5/speech-dispatcher-0.11.5.tar.gz
+        sha512: d6d880bce0ae5bc2a5d519ef7740c689ae8b4b0bb658379762810e4beae3e465a429fbe19f7c490e89db0ea6a36aedd4b2287ac9251b90059b5c2cb3c0dd8a28
+    modules:
+      # dotconf provides utility functions to parse config files. It's only linked with the speech-dispatcher server,
+      # which we aren't building, but it needs to exist to get past the ./configure step anyway
+      - name: dotconf
+        sources:
```
Comment (bbhtt) [inline org.electronjs.Electron2.BaseApp.yml:127]: add a cleanup, this is build time only, `"*"`

File: org.electronjs.Electron2.BaseApp.yml [hunk 3]
```diff
@@ -94,3 +94,46 @@ modules:
...
+      - --without-flite
+      - --disable-python
+      - --with-systemdsystemunitdir=/app/lib/systemd/system/speech-dispatcherd.service
+    no-make-install: true
+    post-install:
+      - cd ./src/api/c && make install
+    cleanup:
+      - "*.la"
+      - "*.a"
+      - /include
```
Comment (bbhtt) [inline org.electronjs.Electron2.BaseApp.yml:118]: Does electron need the base library libspeechd.so only If so further things should be cleaned up https://github.com/flathub/org.mozilla.firefox.BaseApp/blob/7c7fe8222b3066786868174f9462409122ac123b/org.mozilla.firefox.BaseApp.json#L139-L146

Comment (bbhtt) [reply org.electronjs.Electron2.BaseApp.yml:118]: > [Chromium source code for proof](https://source.chromium.org/chromium/chromium/src/+/refs/tags/122.0.6261.111:content/browser/speech/tts_linux.cc;l=251).

Seems like so, please cleanup rest of the stuff

File: org.electronjs.Electron2.BaseApp.yml [hunk 4]
```diff
@@ -94,3 +94,46 @@ modules:
...
+      - --with-ibmtts=no
+      - --with-kali=no
+      - --with-baratinoo=no
+      - --with-voxin=no
+      - --without-flite
+      - --disable-python
+      - --with-systemdsystemunitdir=/app/lib/systemd/system/speech-dispatcherd.service
+    no-make-install: true
+    post-install:
+      - cd ./src/api/c && make install
```
Comment (bbhtt) [inline org.electronjs.Electron2.BaseApp.yml:114]: This should not be needed. See the example in the Firefox baseapp.

Comment (hypevhs) [reply org.electronjs.Electron2.BaseApp.yml:114]: This builds specifically the C API, the main artifact of which is `libspeechd.so`. This is opposed to building the entire speech-dispatcher suite which includes the daemon we aren't using.

We're building a subset of speech-dispatcher here. Should we instead build everything, and just add more entries to the cleanup section?

Comment (bbhtt) [reply org.electronjs.Electron2.BaseApp.yml:114]: Hmm seems fine, I didn't see the directory.

File: org.electronjs.Electron2.BaseApp.yml (hunk 3 above)
Comment (bbhtt) [reply org.electronjs.Electron2.BaseApp.yml:118]: Not needed.

File: org.electronjs.Electron2.BaseApp.yml [hunk 5]
```diff
@@ -94,3 +94,46 @@ modules:
...
+            url: https://github.com/williamh/dotconf/archive/refs/tags/v1.4.1.tar.gz
+            sha512: a6cada8621295b268d4b4fd85bc0c207e78324c9e84754ead2fdf6c1598ec8bdf626f9c24e66063d921c95d73e83b50ab50416a9b4c9a7a631392552ec46f55a
+          - type: script
+            commands:
+              - autoreconf -fiv
+            dest-filename: autogen.sh
+        cleanup:
+          - /include
+          - /lib/pkgconfig
+          - /share/doc
```
Comment (bbhtt) [inline org.electronjs.Electron2.BaseApp.yml]:
```suggestion
        cleanup:
          - "*"
```

This is needed at build time only.

Comment (hypevhs) [reply org.electronjs.Electron2.BaseApp.yml]: Added

File: org.electronjs.Electron2.BaseApp.yml (hunk 1 above)
Comment (hypevhs) [reply org.electronjs.Electron2.BaseApp.yml]: I was able to just remove the flag.

**Comment by bbhtt** [comment]
bot, build

**Comment by bbhtt** [comment]
Thanks.

**Pull request merged**